File: java_sdk/src/main/java/business_api_client/PixeleventcreatePixelEvents.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.PixeleventcreateRules;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * PixeleventcreatePixelEvents
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-05-20T12:00:30.004772-07:00[America/Los_Angeles]")
public class PixeleventcreatePixelEvents {
  @JsonProperty("currency")
  private String currency = null;

  @JsonProperty("currency_value")
  private String currencyValue = null;

  @JsonProperty("event_code")
  private String eventCode = null;

  @JsonProperty("event_id")
  private String eventId = null;

  @JsonProperty("event_name")
  private String eventName = null;

  @JsonProperty("event_type")
  private String eventType = null;

  @JsonProperty("rules")
  private List<PixeleventcreateRules> rules = null;

  @JsonProperty("statistic_type")
  private String statisticType = null;

  public PixeleventcreatePixelEvents currency(String currency) {
    this.currency = currency;
    return this;
  }

   /**
   * Get currency
   * @return currency
  **/
  @Schema(description = "")
  public String getCurrency() {
    return currency;
  }

  public void setCurrency(String currency) {
    this.currency = currency;
  }

  public PixeleventcreatePixelEvents currencyValue(String currencyValue) {
    this.currencyValue = currencyValue;
    return this;
  }

   /**
   * Get currencyValue
   * @return currencyValue
  **/
  @Schema(description = "")
  public String getCurrencyValue() {
    return currencyValue;
  }

  public void setCurrencyValue(String currencyValue) {
    this.currencyValue = currencyValue;
  }

  public PixeleventcreatePixelEvents eventCode(String eventCode) {
    this.eventCode = eventCode;
    return this;
  }

   /**
   * Get eventCode
   * @return eventCode
  **/
  @Schema(description = "")
  public String getEventCode() {
    return eventCode;
  }

  public void setEventCode(String eventCode) {
    this.eventCode = eventCode;
  }

  public PixeleventcreatePixelEvents eventId(String eventId) {
    this.eventId = eventId;
    return this;
  }

   /**
   * Get eventId
   * @return eventId
  **/
  @Schema(description = "")
  public String getEventId() {
    return eventId;
  }

  public void setEventId(String eventId) {
    this.eventId = eventId;
  }

  public PixeleventcreatePixelEvents eventName(String eventName) {
    this.eventName = eventName;
    return this;
  }

   /**
   * Get eventName
   * @return eventName
  **/
  @Schema(description = "")
  public String getEventName() {
    return eventName;
  }

  public void setEventName(String eventName) {
    this.eventName = eventName;
  }

  public PixeleventcreatePixelEvents eventType(String eventType) {
    this.eventType = eventType;
    return this;
  }

   /**
   * Get eventType
   * @return eventType
  **/
  @Schema(description = "")
  public String getEventType() {
    return eventType;
  }

  public void setEventType(String eventType) {
    this.eventType = eventType;
  }

  public PixeleventcreatePixelEvents rules(List<PixeleventcreateRules> rules) {
    this.rules = rules;
    return this;
  }

  public PixeleventcreatePixelEvents addRulesItem(PixeleventcreateRules rulesItem) {
    if (this.rules == null) {
      this.rules = new ArrayList<PixeleventcreateRules>();
    }
    this.rules.add(rulesItem);
    return this;
  }

   /**
   * Get rules
   * @return rules
  **/
  @Schema(description = "")
  public List<PixeleventcreateRules> getRules() {
    return rules;
  }

  public void setRules(List<PixeleventcreateRules> rules) {
    this.rules = rules;
  }

  public PixeleventcreatePixelEvents statisticType(String statisticType) {
    this.statisticType = statisticType;
    return this;
  }

   /**
   * Get statisticType
   * @return statisticType
  **/
  @Schema(description = "")
  public String getStatisticType() {
    return statisticType;
  }

  public void setStatisticType(String statisticType) {
    this.statisticType = statisticType;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PixeleventcreatePixelEvents pixeleventcreatePixelEvents = (PixeleventcreatePixelEvents) o;
    return Objects.equals(this.currency, pixeleventcreatePixelEvents.currency) &&
        Objects.equals(this.currencyValue, pixeleventcreatePixelEvents.currencyValue) &&
        Objects.equals(this.eventCode, pixeleventcreatePixelEvents.eventCode) &&
        Objects.equals(this.eventId, pixeleventcreatePixelEvents.eventId) &&
        Objects.equals(this.eventName, pixeleventcreatePixelEvents.eventName) &&
        Objects.equals(this.eventType, pixeleventcreatePixelEvents.eventType) &&
        Objects.equals(this.rules, pixeleventcreatePixelEvents.rules) &&
        Objects.equals(this.statisticType, pixeleventcreatePixelEvents.statisticType);
  }

  @Override
  public int hashCode() {
    return Objects.hash(currency, currencyValue, eventCode, eventId, eventName, eventType, rules, statisticType);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class PixeleventcreatePixelEvents {\n");
    
    sb.append("    currency: ").append(toIndentedString(currency)).append("\n");
    sb.append("    currencyValue: ").append(toIndentedString(currencyValue)).append("\n");
    sb.append("    eventCode: ").append(toIndentedString(eventCode)).append("\n");
    sb.append("    eventId: ").append(toIndentedString(eventId)).append("\n");
    sb.append("    eventName: ").append(toIndentedString(eventName)).append("\n");
    sb.append("    eventType: ").append(toIndentedString(eventType)).append("\n");
    sb.append("    rules: ").append(toIndentedString(rules)).append("\n");
    sb.append("    statisticType: ").append(toIndentedString(statisticType)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/PixeleventcreatePixelEvents.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {PixeleventcreateRules} from './PixeleventcreateRules.js';

/**
 * The PixeleventcreatePixelEvents model module.
 * @module model/PixeleventcreatePixelEvents
 * @version 0.1.8
 */
export class PixeleventcreatePixelEvents {
  /**
   * Constructs a new <code>PixeleventcreatePixelEvents</code>.
   * @alias module:model/PixeleventcreatePixelEvents
   * @class
   */
  constructor() {
  }

  /**
   * Constructs a <code>PixeleventcreatePixelEvents</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PixeleventcreatePixelEvents} obj Optional instance to populate.
   * @return {module:model/PixeleventcreatePixelEvents} The populated <code>PixeleventcreatePixelEvents</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new PixeleventcreatePixelEvents();
      if (data.hasOwnProperty('currency'))
        obj.currency = ApiClient.convertToType(data['currency'], 'String');
      if (data.hasOwnProperty('currency_value'))
        obj.currency_value = ApiClient.convertToType(data['currency_value'], 'String');
      if (data.hasOwnProperty('event_code'))
        obj.event_code = ApiClient.convertToType(data['event_code'], 'String');
      if (data.hasOwnProperty('event_id'))
        obj.event_id = ApiClient.convertToType(data['event_id'], 'String');
      if (data.hasOwnProperty('event_name'))
        obj.event_name = ApiClient.convertToType(data['event_name'], 'String');
      if (data.hasOwnProperty('event_type'))
        obj.event_type = ApiClient.convertToType(data['event_type'], 'String');
      if (data.hasOwnProperty('rules'))
        obj.rules = ApiClient.convertToType(data['rules'], [PixeleventcreateRules]);
      if (data.hasOwnProperty('statistic_type'))
        obj.statistic_type = ApiClient.convertToType(data['statistic_type'], 'String');
    }
    return obj;
  }
}

/**
 * @member {String} currency
 */
PixeleventcreatePixelEvents.prototype.currency = undefined;

/**
 * @member {String} currency_value
 */
PixeleventcreatePixelEvents.prototype.currency_value = undefined;

/**
 * @member {String} event_code
 */
PixeleventcreatePixelEvents.prototype.event_code = undefined;

/**
 * @member {String} event_id
 */
PixeleventcreatePixelEvents.prototype.event_id = undefined;

/**
 * @member {String} event_name
 */
PixeleventcreatePixelEvents.prototype.event_name = undefined;

/**
 * @member {String} event_type
 */
PixeleventcreatePixelEvents.prototype.event_type = undefined;

/**
 * @member {Array.<module:model/PixeleventcreateRules>} rules
 */
PixeleventcreatePixelEvents.prototype.rules = undefined;

/**
 * @member {String} statistic_type
 */
PixeleventcreatePixelEvents.prototype.statistic_type = undefined;

